xls/modules/apfloat_sub_2.x

```diff
@@ -0,0 +1,263 @@
+// Copyright 2021 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file implements most of IEEE
+// floating-point subtraction (and comparisons that are 
+// implemented using subtraction), with the following exceptions:
+//  - Both input and output denormals are treated as/flushed to 0.
+//  - Only round-to-nearest mode is supported.
+//  - No exception flags are raised/reported.
+// In all other cases, results should be identical to other
+// conforming implementations (modulo exact significand
+// values in the NaN case.
+import xls.dslx.stdlib.apfloat
+import xls.dslx.stdlib.float32
```

Comment:
I'd like to keep the APFloat fn implementations completely independent from any individual instantiation. Would you mind separating the tests out to a separate file (and thus the reference to float32)?

(I'm open to push-back)

Reply:
I switched from float32 references to apfloat references for the tests.
I prefer to have the tests with the apfloat implementations rather with the 32 / 64 / etc bit derivatives - seems "cleaner" / more direct that way.